src/millie/utils.nim

```diff
@@ -1,25 +1,12 @@
-import strutils, strformat, math
-
-## parseNum ensures the value is a number (double)
-## returns the parsed value.
-proc parseNum*(value: SomeNumber): float =
-  var val: float
-  
-  try: val = parseFloat($value)
-  except ValueError: raise newException(ValueError, fmt"Input value must be a number, not '{value}'")
-
-  return val
-
-## roundTo rounds a number up to a specified precision and returns it
-proc roundTo*(value: SomeNumber, precision: int): SomeNumber = return parseFloat($value).round(precision)
-
-## divideTillEnd divides and calls the passed function until `res` is less than 0.
-proc divideTillEnd*(value: SomeNumber, fun: proc (res: float)) =
-  var denom = 1000;
+template divideTillEnd*(value: SomeNumber, code: untyped) =
+  ## divideTillEnd divides the value and executes the passed code
+  ## until `res` is less than 1.
+  let vals = value # save the value
+  var denom = 1000
   while true:
-    var res = value / denom.toFloat
+    var res {.inject.} = vals / denom.toFloat
```

Comment:
`{.inject.}` is here to let the code that we pass to the template to be able to see the variable